app/js/adminPage.js

```diff
@@ -74,51 +80,60 @@ function isEmailValid(email) {
  * @returns {boolean} - Does the user already exist.
  */
 function userExists(emailToAdd, existingUsers) {
-    var result = false
+    let result = false;
     existingUsers.forEach(function(user) {
         if (user.email === emailToAdd) {
-            result = true
+            result = true;
         }
-    })
+    });
 
-    return result
+    return result;
 }
 
 document.querySelector('#addNewUserForm').addEventListener('submit', function(event) {
-    event.preventDefault()
-    let emailField = document.querySelector('#email')
-    let nameField = document.querySelector('#name')
-    let testField = document.querySelector('#test_id')
-    let errorField = document.querySelector('#error')
-    let timeField = document.querySelector('#time')
+    event.preventDefault();
+    let emailField = document.querySelector('#email');
+    let nameField = document.querySelector('#name');
+    let testField = document.querySelector('#test_id');
+    let errorField = document.querySelector('#error');
+    let timeMinutes = parseInt(document.querySelector('#user_time_minutes').value);
+    let timeSeconds = parseInt(document.querySelector('#user_time_seconds').value);
+    let timeTotal = (timeMinutes * 60) + timeSeconds;
 
     getExistingUsers().then(function(existingUsers) {
-
-        let emailIsValid = true
-        let timeIsValid = true
-
         if (!isEmailValid(emailField.value) || userExists(emailField.value, existingUsers)) {
-            emailIsValid = false
-            errorField.innerHTML = "Your email is not valid or already exists: Please provide a correct email"
-        }
-        if (timeField.value <=1 || timeField.value == null || isNaN(timeField.value) === true ) {
-            timeIsValid = false
-            errorField.innerHTML += 'This is not a good number!'
-        }
-
-        if (emailIsValid && timeIsValid) {
-            errorField.innerHTML = ''
-            var setTime = timeField.value * 60
-            saveNewUser({'name': nameField.value, 'email': emailField.value, 'test_id': testField.value, 'time': setTime}).then(function(response) {
+            errorField.classList.remove('alert-success');
+            errorField.classList.add('alert-danger');
+            errorField.innerHTML = "Your email is not valid or already exists: Please provide a correct email";
+        } else if (timeTotal <= 1 || timeTotal > 3600 ||
```

Comment:
theres functions for a lot of this logic in `modal.js` already

also, that is definutly not the right place for those functions